homeassistant/components/hisense_connectlife/climate.py

```diff
@@ -0,0 +1,777 @@
+        if not devices:
+            _LOGGER.warning("No devices found in coordinator data")
+            return
+
+        entities = []
+        for device in devices.values():
+            _LOGGER.debug("Processing device: %s", device.to_dict())
+            if isinstance(device, HisenseDeviceInfo) and device.is_supported():
+                _LOGGER.info(
+                    "Adding climate entity for device: %s (type: %s-%s)",
+                    device.name,
+                    device.type_code,
+                    device.feature_code,
+                )
+                entity = HisenseClimate(coordinator, device)
+                entities.append(entity)
+            else:
+                _LOGGER.warning(
+                    "Skipping unsupported device: %s-%s (%s)",
+                    getattr(device, "type_code", None),
+                    getattr(device, "feature_code", None),
+                    getattr(device, "name", None),
+                )
+
+        if not entities:
+            _LOGGER.warning("No supported devices found")
+            return
+
+        _LOGGER.info("Adding %d climate entities", len(entities))
+        async_add_entities(entities)
+
+    except Exception as err:
+        _LOGGER.error("Failed to set up climate platform: %s", err)
+        raise
+
+
+class HisenseClimate(CoordinatorEntity, ClimateEntity):
+    """Hisense AC climate entity."""
+
+    _attr_has_entity_name = False
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_min_temp = MIN_TEMP
+    _attr_max_temp = MAX_TEMP
+    _attr_target_temperature_step = 1
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.FAN_MODE
+        | ClimateEntityFeature.SWING_MODE
+        | ClimateEntityFeature.TURN_ON
+        | ClimateEntityFeature.TURN_OFF
+    )
+
+    def __init__(
+        self,
+        coordinator: HisenseACPluginDataUpdateCoordinator,
+        device: HisenseDeviceInfo,
+    ) -> None:
+        """Initialize the climate entity."""
+        super().__init__(coordinator)
+        self._device_id = device.puid
+        self._attr_unique_id = f"{device.device_id}_climate"
+        self._attr_name = device.name
+        self.hasAuto = False
+        self._last_command_time = 0
+        self.wait_time = 3
+        self._cached_target_temp = None
+        self._cached_hvac_mode = HVACMode.OFF
+        self._cached_fan_mode = None
+        self._cached_swing_mode = SWING_OFF
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, device.device_id)},
+            name=device.name,
+            manufacturer="Hisense",
+            model=f"{device.type_name} ({device.feature_name})",
+        )
+        if device.feature_code == "19901":
+            self._attr_target_temperature_step = 0.5
+        # Get device parser to determine available modes and options
+        device_type = device.get_device_type()
+        if device_type:
+            try:
```

Comment:
Same comments as before so not commenting on it again, just see for yourself if there are more, but tighten your exceptions and limit try-ing to what could fail. I'm also puzzled by why you'd debug-log an inability to parse, without any feedback and continued processing of other things?